superset/assets/spec/javascripts/visualizations/core/createLoadableRenderer_spec.jsx

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import createLoadableRenderer from 'src/visualizations/core/components/createLoadableRenderer';
+
+describe('createLoadableRenderer', () => {
+  function TestComponent() {
+    return (
+      <div className="test-component">test</div>
+    );
+  }
+  let loadChartSuccess;
+  let render;
+  let loading;
+  let LoadableRenderer;
+
+  beforeEach(() => {
+    loadChartSuccess = jest.fn(() => Promise.resolve(TestComponent));
+    render = jest.fn((loaded) => {
+      const { Chart } = loaded;
+      return (<Chart />);
+    });
+    loading = jest.fn(() => (<div>Loading</div>));
+    LoadableRenderer = createLoadableRenderer({
+      loader: {
+        Chart: loadChartSuccess,
+      },
+      loading,
+      render,
+    });
+  });
+
+  describe('returns a LoadableRenderer class', () => {
+    it('LoadableRenderer.preload() preloads the lazy-load components', () => {
+      expect(LoadableRenderer.preload).toBeInstanceOf(Function);
+      LoadableRenderer.preload();
+      expect(loadChartSuccess).toHaveBeenCalled();
+    });
+
+    it('calls onRenderSuccess when succeeds', (done) => {
+      const onRenderSuccess = jest.fn();
+      const onRenderFailure = jest.fn();
+      shallow(
+        <LoadableRenderer
+          onRenderSuccess={onRenderSuccess}
+          onRenderFailure={onRenderFailure}
+        />,
+      );
+      expect(loadChartSuccess).toHaveBeenCalled();
+      setTimeout(() => {
+        expect(render).toHaveBeenCalled();
+        expect(onRenderSuccess).toHaveBeenCalled();
```

Comment:
would it be useful to add a specific `callCount=1` in these tests? `CustomLoadableRenderer` calls `afterRender` in `componentDidMount` and `componentDidUpdate` so just want to make sure it doesn't call it 2x.

Reply:
👍